### ISSUE TITLE
Why is the dev data set in run_classifier.py not used for model training?

### ISSUE THREAD

**Issue by echan00** [google-research/bert#593]
I'm looking through the run_classifier.py code right now and see only the training data is being used for model training (and only the dev data is being used for evaluation.)

Shouldn't the dev data set also be employed for model training? And a test data set should be used for model evaluation?

**Comment by nellymin** [comment]
>Shouldn't the dev data set also be employed for model training?
Yes, the dev set (usually called validation set) is in fact indirectly used for training when you change up the hyper parameters based on it's accuracy.
Тhe actual accuracy of your model however is calculated using the test set because it has never been used for manipulating the model. Therefore it is unbiased and will prevent you from thinking you have done a good job when you actually have not.
